language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToFunctionCodeAction.java

```diff
@@ -490,6 +501,16 @@ private Optional<ArgListsHolder> getArgLists(CodeActionContext context, List<Sym
         return Optional.of(argListsHolder);
     }
 
+    private Optional<TypeSymbol> getTypeSymbol(CodeActionContext context, TypeSymbol typeSymbol,
+                                               NonTerminalNode node) {
+        // If the type symbol kind is union, return the narrowed type using typeOf() for the particular node.
+        if (typeSymbol.typeKind() == TypeDescKind.UNION) {
+            return context.currentSemanticModel().flatMap(semanticModel ->
+                    semanticModel.typeOf(node.children().get(0)).flatMap(SymbolUtil::getTypeDescriptor));
```

Comment:
Here we shouldn't use `node.children().get(0)`. Instead, let's check the SyntaxKind of `node` and access the child we want using relevant methods.

Reply:
We can check if the SyntaxKind of `node.parent()` is `ConditionalExpressionNode` and then use `((ConditionalExpressionNode)node.parent()).middleExpression().children().get(0)`
WDYT?